core/commonMain/src/ArgParser.kt

```diff
@@ -51,14 +51,16 @@ interface ArgumentValueDelegate<T> {
      *
      * @see CLIEntity.value
      */
-    // TODO: decide on the exception type
-    // TODO: decide whether 'set' is possible before calling 'parse'
     var value: T
 
     /** Provides the value for the delegated property getter. Returns the [value] property. */
     operator fun getValue(thisRef: Any?, property: KProperty<*>): T = value
     /** Sets the [value] to the [ArgumentValueDelegate.value] property from the delegated property setter. */
     operator fun setValue(thisRef: Any?, property: KProperty<*>, value: T) {
+        check((this as ParsingValue<*, *>).valueOrigin != ArgParser.ValueOrigin.UNDEFINED) {
+            "Resetting value of option/argument is only possible after parsing command line arguments." +
+                    " ArgParser.parse(...) method should be called before"
+        }
```

Comment:
This check is just here or it's duplicated in the `value` property setter's actual implementation?
In the former case it is inconsistent, and in case of the latter it seems redundant.